test.js

```diff
@@ -303,7 +302,6 @@ test('Cleans up input streams listeners on merged stream end', async t => {
 	const stream = mergeStreams([inputStream]);
 	stream.end();
 	await stream.toArray();
-	await scheduler.yield();
```

Comment:
All those `await scheduler.yield()` were symptoms of creating race conditions by not making `.remove()` async.